cosmo_tester/test_suites/bootstrap_based_tests/distributed_installation_test.py

```diff
@@ -0,0 +1,159 @@
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+import time
+import pytest
+
+from cosmo_tester.framework.examples.nodecellar import NodeCellarExample
+from cosmo_tester.framework.test_hosts import \
+    DistributedInstallationCloudifyManager
+from cosmo_tester.framework.util import prepare_and_get_test_tenant
+
+from cosmo_tester.test_suites.snapshots import (
+    create_snapshot,
+    restore_snapshot
+)
+
+USER_NAME = "test_user"
+USER_PASS = "testuser123"
+TENANT_NAME = "tenant"
+
+
+@pytest.fixture(scope='module')
+def database_and_manager(cfy, ssh_key, module_tmpdir, attributes, logger):
+    """
+    Bootstraps a cloudify manager with no DB on a VM in rackspace OpenStack.
+    """
+    # The preconfigure callback populates the files structure prior to the BS
+    def _preconfigure_callback_manager_and_database(database_and_manager):
+        # Updating the database VM first
+        database_and_manager[0].additional_install_config = {
+            'sanity': {'skip_sanity': 'true'},
+            'postgresql_server': {'enable_remote_connections': 'true'},
+            'services_to_install': [
+                'database_service'
+            ]
+        }
+        database_and_manager[1].additional_install_config = {
+            'sanity': {'skip_sanity': 'true'},
+            'postgresql_client': {
+                'host': str(database_and_manager[0].private_ip_address)
+            },
+            'services_to_install': [
+                'queue_service',
+                'composer_service',
+                'manager_service'
+            ]
+        }
+
+    hosts = DistributedInstallationCloudifyManager(cfy=cfy,
+                                                   ssh_key=ssh_key,
+                                                   tmpdir=module_tmpdir,
+                                                   attributes=attributes,
+                                                   logger=logger)
+
+    hosts.instances[1].upload_plugins = False
+
+    hosts.preconfigure_callback = _preconfigure_callback_manager_and_database
+
+    try:
+        hosts.create()
+        yield hosts.instances
+    finally:
+        hosts.destroy()
+
+
+@pytest.fixture(scope='function')
+def nodecellar(cfy, database_and_manager, attributes, ssh_key, tmpdir, logger):
+    # Uploading to the manager not the database
+    manager = database_and_manager[1]
+    manager.use()
+    tenant = prepare_and_get_test_tenant(TENANT_NAME, manager, cfy)
+    nc = NodeCellarExample(
+        cfy, manager, attributes, ssh_key, logger, tmpdir,
+        tenant=tenant, suffix='simple')
+    nc.blueprint_file = 'simple-blueprint-with-secrets.yaml'
+    # nc.blueprint_file = 'openstack-blueprint.yaml' # alternative bp for test
+    yield nc
+
+
+def test_distributed_installation_scenario(database_and_manager,
```

Comment:
Maybe call this "test_basic_flow", because the path to the test will relate it to this subject